connectors/anaplan/app/handlers.py

```
@@ -0,0 +1,234 @@
+
+        Validates the provided credentials by attempting to obtain an authentication
+        token from App's authentication service.
+
+        Returns:
+            bool: True if authentication is successful, False otherwise.
+
+        Note:
+            Endpoint: POST /workflows/v1/auth (triggered by UI "Test Connection" button)
+            The SDK wraps the boolean response in {"success": true/false, "message": "..."}.
+        """
+        try:
+            if not self.client:
+                logger.error("App client not initialized")
+                return False
+
+            result = await self.client._get_auth_token()
+            if result:
+                logger.info("Authentication test successful")
+                return True
+            else:
+                logger.error("Authentication test failed")
+                return False
+
+        except Exception as exc:
+            logger.error(f"Failed to authenticate: {str(exc)}")
+            return False
+
+    async def fetch_metadata(self, **kwargs: Any) -> List[Dict[str, Any]]:
+        """Fetch App metadata (apps, pages) for UI filter display.
+
+        Retrieves all available apps and their associated pages from Anaplan
+        and formats them in a hierarchical structure for the UI dropdowns.
+
+        Args:
+            **kwargs: Additional keyword arguments (metadata_type and database
+                are ignored for App).
+
+        Returns:
+            List[Dict[str, Any]]: Hierarchical metadata structure with apps and pages.
+
+        Raises:
+            Exception: If client is not initialized or metadata fetching fails.
+
+        Note:
+            Endpoint: POST /workflows/v1/metadata (triggered by UI when user reaches page 3)
+            The SDK wraps the response in {"success": true/false, "data": "..."}.
+        """
+        try:
+            if not self.client:
+                raise Exception("App client not initialized")
```

I think you should raise a specific exception like `application_sdk.common.error_codes.ClientError` not `Exception`